app/src/test/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailScreenTest.kt

```diff
@@ -0,0 +1,178 @@
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagDetailAppbarTitle).assertTextEquals(coinDetailUiModel.coinName)
+            onNodeWithTag(testTag = TestTagDetailLogo).assertIsDisplayed()
+            onNodeWithTag(testTag = TestTagDetailCurrentPrice).assertTextEquals(
+                "$${coinDetailUiModel.currentPrice.toFormattedString()}"
+            )
+            onNodeWithTag(testTag = TestTagDetailPriceChangePercent).assertTextEquals(
+                "${abs(coinDetailUiModel.priceChangePercentage24hInCurrency).toFormattedString()}%"
+            )
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the Chart properly`() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagDetailLineChart).assertExists()
+            onNodeWithTag(testTag = TestTagDetailChartInterval).assertExists()
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the CoinInfo properly`() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagDetailCoinInfo).assertExists()
+
+            with(onAllNodesWithTag(testTag = TestTagDetailItemTitle)) {
+                onFirst().assertTextEquals(marketCapTitle)
+                this[1].assertTextEquals(allTimeHighTitle)
+                onLast().assertTextEquals(allTimeLowTitle)
+            }
+
+            with(onAllNodesWithTag(testTag = TestTagDetailItemPrice)) {
+                onFirst().assertTextEquals("$${coinDetailUiModel.marketCap.toFormattedString()}")
+                this[1].assertTextEquals("$${coinDetailUiModel.ath.toFormattedString()}")
+                onLast().assertTextEquals("$${coinDetailUiModel.atl.toFormattedString()}")
+            }
+
+            with(onAllNodesWithTag(
+                testTag = TestTagDetailItemPriceChange
+            )) {
+                onFirst().onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.marketCapChangePercentage24h).toFormattedString()}%"
+                )
+                this[1].onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.athChangePercentage).toFormattedString()}%"
+                )
+                onLast().onChild().assertTextEquals(
+                    "${abs(coinDetailUiModel.atlChangePercentage).toFormattedString()}%"
+                )
+            }
+        }
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail successfully, it renders the SellBuyGroup properly`() {
+        initViewModel()
+
+        composeAndroidTestRule.onNodeWithTag(testTag = TestTagDetailSellBuyGroup).assertExists()
+    }
+
+    @Test
+    fun `When enter to DetailScreen and GetCoinDetail failed, it shows the Toast properly`() {
```

Comment:
we should ensure that's the error message, we don't need to care about what that UI component is.
```suggestion
    fun `When entering to the DetailScreen and GetCoinDetail failed, it shows the error message properly`() {
```